### PR TITLE
Expected Records to `.jsonl` format

### LINKED ISSUE #20569: Fix integration test files for Spotless
Spotless failed to format next files: 
`airbyte-integrations/connectors/**/integration_tests/expected_records.json`,
With: 
* What went wrong:
```
Execution failed for task ':spotlessStyling'.
> Unexpected response status code at /prettier/format [HTTP 501] -- (Error while formatting: SyntaxError: Unexpected token (2:1)
    1 | {"stream": "sat_basic_dataset", "data": {"id": 1, "test_column_1": -128, "test_column_10": -128, "test_column_11": -128, "test_column_12": -128.0, "test_column_13": true, "test_column_15": "2021-10-20T00:00:00Z", "test_column_16": "2021-10-20T11:22:33Z", "test_column_17": "2021-10-20T11:22:33Z", "test_column_19": "qwe", "test_column_2": -128, "test_column_21": "15:30:00", "test_column_3": -128, "test_column_4": -128, "test_column_5": -128, "test_column_6": -128, "test_column_7": -128, "test_column_8": -128, "test_column_9": -128 }, "emitted_at": 1669643054045 }
  > 2 | {"stream": "sat_basic_dataset", "data": {"id": 2, "test_column_1": 127, "test_column_10": 127, "test_column_11": 127, "test_column_12": 127.0, "test_column_13": false, "test_column_15": "9999-12-31T00:00:00Z", "test_column_16": "9999-12-31T11:22:33Z", "test_column_17": "2021-10-20T11:22:33Z", "test_column_19": "\u0439\u0446\u0443", "test_column_2": 127, "test_column_21": "15:30:00", "test_column_3": 127, "test_column_4": 127, "test_column_5": 127, "test_column_6": 127, "test_column_7": 127, "test_column_8": 127, "test_column_9": 127 }, "emitted_at": 1669643054045 }
      | ^
```

As a workaround it was excluded from Spotless check (https://github.com/airbytehq/airbyte/blob/master/build.gradle#L131 ): 
```
            '**/integration_tests/expected_records.json',
```
### Expected result 
The `./gradlew --no-daemon format --scan` should not fail for `/integration_tests/expected_records.json` files.

### PR DESCRIPTION
**Pull request by evantahler**
## Overview

closes https://github.com/airbytehq/airbyte/issues/20569

`expected_records` should be using the `.jsonl` file extension.  Learn more at https://jsonlines.org 

Using the proper file extension keeps linters happy.

## Testing notes:
* In general things look good!
* I did not test every connector that was modified here... but did a random sample and any connector that was more than just a single rename
* The failure in `Connector Ops CI / Check test strictness level (pull_request)` is known
* The [failure](https://github.com/airbytehq/airbyte/actions/runs/3761905938/jobs/6394109113) from `source-salesforce` is in the local integration test, and I don't think would be effected by these changes
* The [failure](https://github.com/airbytehq/airbyte/actions/runs/3761904160/jobs/6394104664) for `source-gitlab` has something to do with the container crashing.  It should be re-run

### CODE REVIEW

**Comment by evantahler** [comment]
SAT testing notes:
* In general things look good!
* The [failure](https://github.com/airbytehq/airbyte/actions/runs/3761905938/jobs/6394109113) from `source-salesforce` is in the local integration test, and I don't think would be effected by these changes

**Review by suhomud — APPROVED** [review]
Thank you for update!

**Review by grishick — APPROVED** [review]

File: docs/connector-development/testing-connectors/source-acceptance-tests-reference.md [hunk 1]
```diff
@@ -154,28 +154,28 @@ Set `validate_data_points=True` if possible. This validation is going to be enab
...
 
 Note: For date-time we are not checking against compliance against ISO8601 \(and RFC3339 as subset of it\). Since we are using specified format to set database column type on db normalization stage, value should be compliant to bigquery [timestamp](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#timestamp_type) and SQL "timestamp with timezone" formats.
 
-### Example of `expected_records.txt`:
+### Example of `expected_records.jsonl`:
 
-In general, the expected\_records.json should contain the subset of output of the records of particular stream you need to test. The required fields are: `stream, data, emitted_at`
+In general, the expected_records.json should contain the subset of output of the records of particular stream you need to test. The required fields are: `stream, data, emitted_at`
```
Comment (pedroslopez) [inline docs/connector-development/testing-connectors/source-acceptance-tests-reference.md:178]:
```suggestion
In general, the expected_records.jsonl should contain the subset of output of the records of particular stream you need to test. The required fields are: `stream, data, emitted_at`
```

**Review by pedroslopez — APPROVED** [review]

**Comment by pedroslopez** [comment]
/test connector=connectors/source-smartsheets

saw that this didn't post the results for whatever reason when this ran before

**Comment by pedroslopez** [comment]
/test connector=connectors/source-snowflake

also saw that this didn't post the results for whatever reason when this ran before

**Review by sherifnada — APPROVED** [review]

**Pull request merged**